src/main/scala/io/movies/interpreters/RepositoryImpl.scala

```diff
@@ -0,0 +1,37 @@
+package io.movies.interpreters
+
+import cats.effect._
+import doobie._
+import doobie.hikari._
+import doobie.implicits._
+import io.movies.algebras.Repository
+import io.movies.model.{Movie, RegisteredMovie}
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+
+object RepositoryImpl {
+
+  //no se pasa el recurso para que no se crea/destruya cada vez. no es viable en el caso de conexiones a BBDD
+  def doobie[F[_]](transactor: HikariTransactor[F])(implicit S: Sync[F], logger: SelfAwareStructuredLogger[F]) : F[Repository[F]] =
+    S.pure(new Repository[F] {
+
+    override def addMovie(movie: Movie): F[Int] = {
+      val insert: Update0 = sql"insert into movie (title, director, year) values (${movie.title}, ${movie.director}, ${movie.releaseDate})".update
+      logger.info(s"add movie $movie")
+      insert.run.transact(transactor)
```

Comment:
**Remember**: You should compose those expressions with `flatMap` or a for-comprehension to perform both actions. Otherwise, you will only run the last expression.

Reply:
As you want to execute those expressions sequentially, but the second expression doesn't need the result of the first one, you could also use the `productR` method or its alias `*>` provided by the `Apply` typeclass:
```
logger.info(s"add movie $movie") *> insert.run.transact(transactor)
```
Under the hood, what `productR` does is:
```
expr1.flatMap(_ => expr2)
```

Reply:
See https://github.com/gutiory/MoviesServer/pull/1/commits/f9633eb9216d0b7d638e27446b60139f1d684dfc